internal/expr/parser.go

```diff
@@ -109,6 +109,62 @@ func (p *Parser) add(part queryPart) {
 	p.partStart = p.pos
 }
 
+// removeComments passes over the input removing comments.
+// It does not allow for nested comments.
+func (p *Parser) removeComments() error {
+	var res bytes.Buffer
+	var textStart = p.pos
+	var textEnd = p.pos
+
+	for p.pos < len(p.input) {
+		textEnd = p.pos
+		// Ignore comment chars in string literals.
+		if ok, err := p.skipStringLiteral(); err != nil {
+			return err
+		} else if ok {
+			continue
+		}
+
+		if p.skipByte('-') || p.skipByte('/') {
+			startPos := p.pos - 1
+			c := p.input[startPos]
+			var end byte
+			if (c == '-' && p.skipByte('-')) || (c == '/' && p.skipByte('*')) {
+				if c == '-' {
+					end = '\n'
```

Comment:
If the line endings are marked by `\r\n`, we'll need to keep and not skip both characters, and write them in the `res` buffer. Of course this is needed if such system are to be supported :) 
Do we need to support those with `\r` only?